connectors/sources/graphql.py

```diff
@@ -0,0 +1,449 @@
+        self.graphql_object_list = self.configuration["graphql_object_list"]
+        self.http_method = self.configuration["http_method"]
+        self.authentication_method = self.configuration["authentication_method"]
+        self.pagination_enabled = self.configuration["pagination_enabled"]
+        self.variables = {}
+        self.headers = {}
+
+    def set_logger(self, logger_):
+        self._logger = logger_
+
+    @cached_property
+    def session(self):
+        timeout = aiohttp.ClientTimeout(total=self.configuration["connection_timeout"])
+        if self.authentication_method == "bearer":
+            self.headers.update(
+                {
+                    "Authorization": f"Bearer {self.configuration['token']}",
+                }
+            )
+            return aiohttp.ClientSession(
+                headers=self.headers,
+                timeout=timeout,
+                raise_for_status=True,
+            )
+        elif self.authentication_method == "basic":
+            basic_auth = aiohttp.BasicAuth(
+                login=self.configuration["username"],
+                password=self.configuration["password"],
+            )
+            return aiohttp.ClientSession(
+                auth=basic_auth,
+                headers=self.headers,
+                timeout=timeout,
+                raise_for_status=True,
+            )
+        else:
+            return aiohttp.ClientSession(
+                headers=self.headers,
+                timeout=timeout,
+                raise_for_status=True,
+            )
+
+    def extract_graphql_data_items(self, key, data):
+        """Returns sub objects from the response based on graphql_object_list
+
+        Args:
+            key (string): Key of data.
+            data (dict): data to extract
+
+        Yields:
+            dictionary/list: Documents from the response
+        """
+        if key in self.graphql_object_list:
+            yield data
+        if isinstance(data, dict):
+            for key, item in data.items():
+                yield from self.extract_graphql_data_items(key=key, data=item)
+        if isinstance(data, list):
+            for doc in data:
+                yield from self.extract_graphql_data_items(key=None, data=doc)
+
+    async def paginated_call(self, graphql_query):
+        ast = parse(graphql_query)  # pyright: ignore
+        visitor = FieldVisitor()
+        visit(ast, visitor)  # pyright: ignore
+
+        for graphql_object in self.graphql_object_list:
+            if not (
+                (
+                    {"after"}.issubset(set(visitor.fields_dict.get(graphql_object, [])))
+                    and "pageInfo" in visitor.fields_dict.keys()
+                )
+                or {"offset", "limit"}.issubset(
+                    set(visitor.fields_dict.get(graphql_object, []))
+                )
+            ):
+                msg = "Pagination is Enabled. Please add fields and variables related to pagination."
```

Comment:
Can we be explicit about what variables we are missing? 

If user would select the pagination type (offset or cursor) in the configuration it will be easier to communicate at this step what variables are missing from the query